rust/datafusion/src/execution/dataframe_impl.rs

```diff
@@ -113,12 +113,7 @@ impl DataFrame for DataFrameImpl {
         right_cols: &[&str],
     ) -> Result<Arc<dyn DataFrame>> {
         let plan = LogicalPlanBuilder::from(&self.plan)
-            .join(
-                Arc::new(right.to_logical_plan()),
-                join_type,
-                left_cols,
-                right_cols,
-            )?
+            .join(&right.to_logical_plan(), join_type, left_cols, right_cols)?
```

Comment:
this is unrelated: it is just a small simplification to the signature of `LogicalPlanBuilder::join` to not require the `Arc`.